db/dcrpg/internal/addrstmts.go

```diff
@@ -34,76 +30,80 @@ const (
 	CreateAddressTable = `CREATE TABLE IF NOT EXISTS addresses (
 		id SERIAL8 PRIMARY KEY,
 		address TEXT,
-		funding_tx_row_id INT8,
-		funding_tx_hash TEXT,
-		funding_tx_vout_index INT8,
-		vout_row_id INT8,
+		tx_hash TEXT,
+		matching_tx_hash TEXT,
 		value INT8,
-		spending_tx_row_id INT8,
-		spending_tx_hash TEXT,
-		spending_tx_vin_index INT4,
-		vin_row_id INT8
-	);`
-
-	SelectAddressAllByAddress = `SELECT * FROM addresses WHERE address=$1 order by id desc;`
-	SelectAddressRecvCount    = `SELECT COUNT(*) FROM addresses WHERE address=$1;`
-
-	SelectAddressUnspentCountAndValue = `SELECT COUNT(*), SUM(value) FROM addresses WHERE address=$1 and spending_tx_row_id IS NULL;`
-	SelectAddressSpentCountAndValue   = `SELECT COUNT(*), SUM(value) FROM addresses WHERE address=$1 and spending_tx_row_id IS NOT NULL;`
-
-	SelectAddressUnspentWithTxn = `SELECT
-									addresses.address,
-									addresses.funding_tx_hash,
-									addresses.value,
-									transactions.block_height,
-									transactions.block_hash
-									FROM addresses 
-									JOIN transactions ON 
-									addresses.funding_tx_hash = transactions.tx_hash 
-									WHERE 
-									addresses.address=$1 
-									AND 
-									addresses.spending_tx_row_id IS NULL`
-
-	SelectAddressLimitNByAddress = `SELECT * FROM addresses WHERE address=$1 order by id desc limit $2 offset $3;`
-
-	SelectAddressLimitNByAddressSubQry = `WITH these as (SELECT * FROM addresses WHERE address=$1)
-		SELECT * FROM these order by id desc limit $2 offset $3;`
-
-	// SelectAddressDebitsLimitNByAddress = `SELECT *
-	// 	FROM addresses
-	// 	WHERE address=$1 AND spending_tx_row_id IS NOT NULL
-	// 	ORDER BY id DESC LIMIT $2 OFFSET $3;`
-	SelectAddressDebitsLimitNByAddress = `WITH these as (SELECT * FROM addresses WHERE address=$1)
-		SELECT * FROM these WHERE spending_tx_row_id IS NOT NULL
-		ORDER BY id DESC LIMIT $2 OFFSET $3;`
-	SelectAddressCreditsLimitNByAddress = `SELECT id, funding_tx_row_id, funding_tx_hash, funding_tx_vout_index, vout_row_id, value
-		FROM addresses
-		WHERE address=$1
-		ORDER BY id DESC LIMIT $2 OFFSET $3;`
-
-	SelectAddressIDsByFundingOutpoint = `SELECT id, address FROM addresses
-		WHERE funding_tx_hash=$1 and funding_tx_vout_index=$2;`
-	SelectAddressIDByVoutIDAddress = `SELECT id FROM addresses
-		WHERE address=$1 and vout_row_id=$2
-		ORDER BY id DESC;`
-
-	SetAddressSpendingForID = `UPDATE addresses SET spending_tx_row_id = $2, 
-		spending_tx_hash = $3, spending_tx_vin_index = $4, vin_row_id = $5 
-		WHERE id=$1;`
-	SetAddressSpendingForOutpoint = `UPDATE addresses SET spending_tx_row_id = $3, 
-		spending_tx_hash = $4, spending_tx_vin_index = $5, vin_row_id = $6 
-		WHERE funding_tx_hash=$1 and funding_tx_vout_index=$2;`
```

Comment:
Is it possible that `SetAddressSpendingForOutpoint` would update multiple rows often?

Reply:
Extremely rarely, and it would not be incorrect to do so.

Reply:
@dmigwi should i have it rebuild my entire db up again with this fix so I can keep testing?

Reply:
@papacarp What I did to save on time was, to dump the state of the postgres db immediately after all the blocks were loaded. Not much was changed on how blocks are loaded thus its safe to restart the data migration from that state with all blocks loaded but not yet started updating addresses table from vins table content.

Command to dump and load the database

```bash
// Dump the postgres database on the shell
pg_dump -U {db_user} {db_name} > dbexport.pgsql

// Command to load the db
psql -U  {db_user} {db_name} < dbexport.pgsql
```